Src/RhinoPlugin.Tests.Xunit/XunitTestFixture.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Microsoft.Win32;
+using System;
+using System.IO;
+using System.Reflection;
+using Xunit;
+
+namespace RhinoPlugin.Tests.Xunit
+{
+
+    /// <summary>
+    /// Shared test context across unit tests that loads rhinocommon.dll and grasshopper.dll
+    /// </summary>
+    public class XunitTestFixture : IDisposable
+    {
+        bool initialized = false;
+        static string rhinoDir;
+        Rhino.Runtime.InProcess.RhinoCore _rhinoCore;
+
+        /// <summary>
+        /// Empty Constuctor
+        /// </summary>
+        public XunitTestFixture()
+        {
+            //get the correct rhino 7 installation directory
+            rhinoDir = Registry.GetValue(@"HKEY_LOCAL_MACHINE\SOFTWARE\McNeel\Rhinoceros\7.0\Install", "Path", null) as string ?? string.Empty;
+            Assert.True(Directory.Exists(rhinoDir), String.Format("Rhino system dir not found: {0}", rhinoDir));
+
+            // Make sure we are running the tests as 64x
+            Assert.True(Environment.Is64BitProcess, "Tests must be run as x64");
+
+            if (initialized)
+            {
+                throw new InvalidOperationException("Initialize Rhino.Inside once");
+            }
+            else
+            {
+                RhinoInside.Resolver.Initialize();
+                initialized = true;
+            }
+
+            // Set path to rhino system directory
+            string envPath = Environment.GetEnvironmentVariable("path");
+            Environment.SetEnvironmentVariable("path", envPath + ";" + rhinoDir);
+
+            // Start a headless rhino instance using Rhino.Inside
+            StartRhino();
+
+            // We have to load grasshopper.dll on the current AppDomain manually for some reason
+            AppDomain.CurrentDomain.AssemblyResolve += ResolveGrasshopper;
+            
+        }
+
+        /// <summary>
+        /// Starting Rhino - loading the relevant libraries
+        /// </summary>
+        [STAThread]
+        public void StartRhino()
+        {
+            _rhinoCore = new Rhino.Runtime.InProcess.RhinoCore(null, Rhino.Runtime.InProcess.WindowStyle.NoWindow);
+        }
+
+        /// <summary>
+        /// Add Grasshopper.dll to the current up domain
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static Assembly ResolveGrasshopper(object sender, ResolveEventArgs args)
```

Comment:
As per previous comment about dropping the static `rhinoDir`
```suggestion
        private Assembly ResolveGrasshopper(object sender, ResolveEventArgs args)
```